src/sources/http_server.rs

```diff
@@ -330,8 +350,23 @@ impl SourceConfig for SimpleHttpConfig {
         let decoder = self.get_decoding_config()?.build()?;
         let log_namespace = cx.log_namespace(self.log_namespace);
 
+        let (glob_headers, mut headers): (Vec<String>, Vec<String>) = self.headers
+            .clone()
+            .into_iter()
+            .partition(|s| s.contains('*'));
+
+        let glob_headers: Vec<glob::Pattern> = remove_duplicates(glob_headers, "headers")
+            .into_iter()
+            .map(|s| glob::Pattern::new(&s).unwrap())
+            .collect();
+
+        if !glob_headers.is_empty() {
+            headers = remove_glob_matches(&headers, &glob_headers, "headers");
+        }
+
         let source = SimpleHttpSource {
-            headers: remove_duplicates(self.headers.clone(), "headers"),
+            headers: remove_duplicates(headers, "headers"),
+            glob_headers: glob_headers.clone(),
```

Comment:
👋  Thanks for adjusting the approach @sonnens 

The current design here of cloning the headers to create a separate list of glob headers is I think a different direction than what @StephenWakely had in mind and adds some storage and processing that should be avoidable.

I think the idea was to use the glob pattern matching in the `enrich_events()` function directly. i.e not having separate lists, but just using the `headers` config list as-is. Additionally, I think Stephens suggestion to check if the supplied headers contain any wildcard as a gate on using the glob pattern matching is a good idea to reduce overhead when wildcards are not used.

Some pseudo code that would go in enrich_events:

```
if any header in self.headers contains "*" {
  // use glob pattern matching to determine which headers to add to the event metadata
else {
  // use headers_config.get
}
```

Some other general notes:
- we should adjust the doc comment in `SimpleHttpConfig` for the `headers` member, to describe this new behavior. That doc comment is used to automatically generate our documentation on the website. 
- we should add a couple docs::examples for the wildcard case
- unit tests should be added to confirm the behavior works as expected.

Reply:
My thought process for splitting in to headers/glob_headers is that HeaderMap.get is cheap but glob.match is much more expensive, and unless I'm misunderstanding the code (entirely probable) there's only one SimpleHttpConfig instantiated per source on startup, so it can eat a few clock cycles and a size_t there in order to save a bunch on every ingest event.

But thinking about it now yeah you're probably right that any wildcard headers are going to mean iterating through all the headers anyway (except in the edge case of someone declaring the entire set of "standard" http headers and then having a couple additional ones or some such thing.) I'll update it to use a boolean "has any wildcard headers" value (and add docs and tests)